packages/playwright/src/common/config.ts

```diff
@@ -108,7 +108,7 @@ export class FullConfigInternal {
       updateSnapshots: takeFirst(configCLIOverrides.updateSnapshots, userConfig.updateSnapshots, 'missing'),
       updateSourceMethod: takeFirst(configCLIOverrides.updateSourceMethod, userConfig.updateSourceMethod, 'patch'),
       version: require('../../package.json').version,
-      workers: 0,
+      workers: resolveWorkers(takeFirst(configCLIOverrides.debug ? 1 : undefined, configCLIOverrides.workers, userConfig.workers, '50%')),
```

Comment:
I think you should check for debug mode in resolveWorkers to make sure it does not go above 1 in project.